hi-ml-histopathology/testhisto/testhisto/datasets/test_panda_slides.py

```diff
@@ -0,0 +1,76 @@
+#  -------------------------------------------------------------------------------------------
+#  Copyright (c) Microsoft Corporation. All rights reserved.
+#  Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+#  -------------------------------------------------------------------------------------------
+
+import shutil
+from pathlib import Path
+from typing import Any
+
+import numpy as np
+import torch
+from pytorch_lightning import seed_everything
+from testhisto.mocks.base_data_generator import MockHistoDataType
+from testhisto.mocks.slides_generator import MockPandaSlidesGenerator, TilesPositioningType
+
+from health_ml.utils.common_utils import seed_monai_if_available
+from histopathology.configs.classification.DeepSMILESlidesPandaBenchmark import DeepSMILESlidesPandaBenchmark
+from histopathology.utils.naming import SlideKey
+
+
+def test_panda_reproducibility(tmp_path: Path) -> None:
+    """Check if subsequent enumerations of the Panda dataset produce identical sequences of tiles."""
+    seed_everything(seed=123, workers=True)
+    seed_monai_if_available(seed=42)
+    tile_size = 28
+    num_tiles = 4
+    wsi_generator = MockPandaSlidesGenerator(
+        dest_data_path=tmp_path,
+        mock_type=MockHistoDataType.FAKE,
+        n_tiles=num_tiles,
+        n_slides=10,
+        n_channels=3,
+        n_levels=3,
+        tile_size=tile_size,
+        background_val=255,
+        tiles_pos_type=TilesPositioningType.RANDOM
+    )
+    wsi_generator.generate_mock_histo_data()
+
+    container = DeepSMILESlidesPandaBenchmark()
+    container.tile_size = tile_size
+    container.max_bag_size = num_tiles
+    container.local_datasets = [tmp_path]
+
+    def get_items_from_new_loader(num_items: int) -> Any:
+        data_module = container.get_data_module()
+        data_module.dataloader_kwargs = {**data_module.dataloader_kwargs,
+                                         "multiprocessing_context": None,
+                                         "num_workers": 0}
+        loader = data_module.train_dataloader()
```

Comment:
You might want to test the val_dataloader and test_dataloader as well especially since we apply different transforms. It's true that we don't apply any random transforms for val and test but testing it might help us catch any source of randomness during inference (e.g if someone accidently uses a random transform or shuffle the val/test dataloaders without properly fixing the seeds).